src/plugins/kibana/public/settings/sections/indices/partials/_edit_field_type.html

```diff
@@ -0,0 +1,13 @@
+<select ng-if="knownFieldTypes[field.name] !== 'geo_point'" name="field_type" ng-model="field.type">
```

Comment:
So the [geoip processor](https://github.com/elastic/elasticsearch/blob/master/docs/plugins/ingest-geoip.asciidoc) is a special case, because its target field must be classified as a geo_point type in the index template and index pattern. That's why I'm doing a special check [here](https://github.com/elastic/kibana/pull/5892/files#diff-5800732e2289cf27b0ca429e629c1bc7R87) for any geoip processors and then preventing any changes to that target field's type.

Other than that, the user can change anything they want. For instance, they might have lat/lon coordinates in their data which they could create their own geo_point field out of without using the geoip processor. In that case they'd need to manually update the type of that field to geo_point.